### PR TITLE
tax rewards distribute to treasures

### PR DESCRIPTION
**Pull request by BananaLF**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: libs/cosmos-sdk/x/mint/client/cli/tx.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"strings"
+)
+
+// GetCmdManageContractMethodBlockedListProposal implements a command handler for submitting a manage contract blocked list
+// proposal transaction
+func GetCmdManageTreasuresProposal(cdc *codec.Codec) *cobra.Command {
+	return &cobra.Command{
+		Use:   "treasures [proposal-file]",
+		Args:  cobra.ExactArgs(1),
+		Short: "Submit an update contract method blocked list proposal",
```
Comment (xiangjianmeng) [inline libs/cosmos-sdk/x/mint/client/cli/tx.go]: Description does not match function

File: libs/cosmos-sdk/x/mint/client/cli/tx.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	"github.com/okex/exchain/libs/cosmos-sdk/x/auth"
+	"github.com/okex/exchain/libs/cosmos-sdk/x/auth/client/utils"
+	utils2 "github.com/okex/exchain/libs/cosmos-sdk/x/mint/client/utils"
+	types2 "github.com/okex/exchain/libs/cosmos-sdk/x/mint/internal/types"
+	"github.com/okex/exchain/x/gov"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+// GetCmdManageContractMethodBlockedListProposal implements a command handler for submitting a manage contract blocked list
```
Comment (xiangjianmeng) [inline libs/cosmos-sdk/x/mint/client/cli/tx.go]: Description does not match function

File: libs/cosmos-sdk/x/mint/internal/keeper/keeper_okchain.go [hunk 3]
```diff
@@ -58,12 +58,70 @@ func (k Keeper) SetOriginalMintedPerBlock(originalMintedPerBlock sdk.Dec) {
...
 		panic("init tokens per block must be non-negative")
 	}
 
 	return nil
-}
\ No newline at end of file
+}
+
+// SetTreasure set the treasures to db
+func (k Keeper) SetTreasure(ctx sdk.Context, treasures []types.Treasure) {
```
Comment (xiangjianmeng) [inline libs/cosmos-sdk/x/mint/internal/keeper/keeper_okchain.go]: SetTreasures

File: libs/cosmos-sdk/x/mint/internal/keeper/keeper_okchain.go [hunk 4]
```diff
@@ -58,12 +58,70 @@ func (k Keeper) SetOriginalMintedPerBlock(originalMintedPerBlock sdk.Dec) {
...
+	store := ctx.KVStore(k.storeKey)
+	types.SortTreasures(treasures)
+	b := k.cdc.MustMarshalBinaryLengthPrefixed(treasures)
+	store.Set(types.TreasuresKey, b)
+}
+
+// GetTreasure get the treasures from db
+func (k Keeper) GetTreasure(ctx sdk.Context) (treasures []types.Treasure) {
```
Comment (xiangjianmeng) [inline libs/cosmos-sdk/x/mint/internal/keeper/keeper_okchain.go]: GetTreasures

File: libs/cosmos-sdk/x/mint/internal/keeper/keeper_okchain.go [hunk 5]
```diff
@@ -65,3 +66,62 @@ func ValidateOriginalMintedPerBlock(originalMintedPerBlock sdk.Dec) error {
...
+	if b != nil {
+		k.cdc.MustUnmarshalBinaryLengthPrefixed(b, &treasures)
+	}
+	return
+}
+
+// AllocateTokenToTreasure allocate token to treasure and return remain
+func (k Keeper) AllocateTokenToTreasure(ctx sdk.Context, fees sdk.Coins) (remain sdk.Coins, err error) {
+	treasures := k.GetTreasures(ctx)
+	remain = remain.Add(fees...)
```
Comment (ilovers) [inline libs/cosmos-sdk/x/mint/internal/keeper/keeper_okchain.go:92]: Receiver 'remain' may be 'nil' in call?

**Review by ilovers — APPROVED** [review]

**Review by xiangjianmeng — APPROVED** [review]

**Pull request merged**